src/ORM/Connect/DBSchemaManager.php

```diff
@@ -640,8 +650,15 @@ public function requireField($table, $field, $spec)
             $specValue = $this->IdColumn(true);
         }
 
+        // Cache of fields
+        static $fieldListCache = [];
+
         if (!$newTable) {
-            $fieldList = $this->fieldList($table);
+            if (!isset($fieldListCache[$table])) {
+                $fieldListCache[$table] = $this->fieldList($table);
```

Comment:
On a “blank” install (framework + cms + asset-admin), this + the index list cache above saves around 300 queries per `dev/build` - we’re currently running a query to fetch a full list of fields for _each_ field we require.